### PR TITLE
Update to 2018 edition idioms

### PR DESCRIPTION
**Pull request by DemiMarie-parity**
As requested by @niklasad1.

### CODE REVIEW

File: uint/tests/uint_tests.rs [hunk 1]
```diff
@@ -8,7 +14,7 @@ extern crate uint;
 extern crate quickcheck;
 
 #[cfg_attr(all(test, feature = "quickcheck"), macro_use(unroll))]
```
Comment (thiolliere) [inline uint/tests/uint_tests.rs]:
```suggestion
```

File: uint/tests/uint_tests.rs [hunk 2]
```diff
@@ -8,7 +14,7 @@ extern crate uint;
 extern crate quickcheck;
```
Comment (thiolliere) [inline uint/tests/uint_tests.rs]: actually macros as well can be import with path, so this could be removed

File: ethbloom/src/lib.rs [hunk 3]
```diff
@@ -140,13 +140,13 @@ impl Bloom {
 	}
 
 	pub fn contains_bloom<'a, B>(&self, bloom: B) -> bool where BloomRef<'a>: From<B> {
-		let bloom_ref: BloomRef = bloom.into();
+		let bloom_ref: BloomRef<'_> = bloom.into();
```
Comment (thiolliere) [inline ethbloom/src/lib.rs]: are all of them needed or just automatically made by a tool ?

Comment (niklasad1) [reply ethbloom/src/lib.rs]: It fixes a warning of `rust 2018 idioms` (warning: hidden lifetime parameters in types are deprecated),  see https://github.com/rust-lang/rust/issues/45992

I think @DemiMarie-parity has executed `cargo fix --edition-idioms`, right?

Comment (DemiMarie-parity) [reply ethbloom/src/lib.rs]: @niklasad1 correct.

File: uint/tests/uint_tests.rs [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Copyright 2015-2017 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
```
Comment (debris) [inline uint/tests/uint_tests.rs]: license is added twice

**Pull request merged**